server/src/main/java/org/elasticsearch/transport/InboundPipeline.java

```diff
@@ -54,13 +55,14 @@ public void close() {
     }
 
     public void handleBytes(TcpChannel channel, ReleasableBytesReference reference) throws IOException {
+        Exception uncaughtException = uncaughtExceptionReference.get();
         if (uncaughtException != null) {
             throw new IllegalStateException("Pipeline state corrupted by uncaught exception", uncaughtException);
```

Comment:
This is the gist of the change.
I opted for this in-line request throwing, because it seemed to me to be the worn path, rather than opt for something such as `ctx.channel().pipeline().fireExceptionCaught(e)` from the eventloop thread.

Reply:
This is no longer the case, we went with the [fireExceptionCaught](https://github.com/elastic/elasticsearch/pull/98814/commits/ee41575a4d905a7eece2c92eb14ff5885a64e681) option.